### PR TITLE
Add optimisation level 3 

### PR DESCRIPTION
**Pull request by sjdilkes**
Add a new optimisation level to `pytket-qiskit` that uses `GreedyPauliSimp` to optimise circuits.

# Checklist

- [x] I have performed a self-review of my code.
- [x] I have commented hard-to-understand parts of my code.
- [x] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: docs/changelog.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 - Update default compilation to use `Qiskit` `SabreLayoutPassManager` as a `CustomPass`.
 - Fix handling of non-default registers when selecting bits in results.
 - The {py:func}`tk_to_qiskit` converter gives a warning if the input {py:class}`~pytket.circuit.Circuit` contains [implicit qubit permutations](https://docs.quantinuum.com/tket/user-guide/manual/manual_circuit.html#implicit-qubit-permutations). 
+- Add new level 3 optimisation that uses `GreedyPauliSimp`
```
Comment (cqc-melf) [inline docs/changelog.md:14]: Can you move this to 0.60.0 ?

File: pytket/extensions/qiskit/backends/aer.py [hunk 2]
```diff
@@ -188,48 +188,67 @@ def _arch_dependent_default_compilation_pass(
...
+                    *arch_specific_passes,
+                    CliffordSimp(False),
+                    SynthesiseTket(),
+                ],
             )
         return SequencePass(
             [
                 DecomposeBoxes(),
-                FullPeepholeOptimise(),
+                RemoveBarriers(),
```
Comment (cqc-melf) [inline pytket/extensions/qiskit/backends/aer.py:205]: Why are you removing the barriers?

File: setup.py [hunk 3]
```diff
@@ -46,7 +46,7 @@
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket >= 1.34.0",
+        "pytket == 1.35.0rc2",
```
Comment (cqc-melf) [inline setup.py:49]: You could do 1.35.0 (please add that to the changelog)

**Review by cqc-melf — COMMENTED** [review]
few small comments

File: setup.py (hunk 3 above)
Comment (sjdilkes) [reply setup.py:49]: Done

File: pytket/extensions/qiskit/backends/aer.py (hunk 2 above)
Comment (sjdilkes) [reply pytket/extensions/qiskit/backends/aer.py:205]: `GreedyPauliSimp` doesn't support them - I'll update the documentation to make it clear that this optimisation level will remove them

**Pull request merged**